frontend/src/components/RideModal/RideModal.tsx

```diff
@@ -1,28 +1,52 @@
 import React, { useEffect, useState } from 'react';
+import moment from 'moment';
 import Modal from '../Modal/Modal';
-import { Button } from '../FormElements/FormElements';
 import { DriverPage, RiderInfoPage, RideTimesPage } from './Pages';
-import { ObjectType } from '../../types/index';
+import { ObjectType, Ride } from '../../types/index';
 import { useReq } from '../../context/req';
 
-const RideModal = () => {
-  const [formData, setFormData] = useState<ObjectType>({});
-  const [isOpen, setIsOpen] = useState(false);
+type RideModalProps = {
+  currentPage: number,
+  setCurrentPage: React.Dispatch<React.SetStateAction<number>>,
+  isOpen: boolean,
+  setIsOpen:  React.Dispatch<React.SetStateAction<boolean>>,
+  setOpenRideModal?: React.Dispatch<React.SetStateAction<number>>,
+  ride?: Ride
+}
```

Comment:
what was the reasoning for moving isOpen to props? i think this might be why the modal has issues closing